app/lib/common/widgets/chat/edit_room_description_sheet.dart

```diff
@@ -0,0 +1,95 @@
+import 'package:acter/common/providers/chat_providers.dart';
+import 'package:acter/common/toolkit/buttons/primary_action_button.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_easyloading/flutter_easyloading.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:flutter_gen/gen_l10n/l10n.dart';
+import 'package:go_router/go_router.dart';
+import 'package:logging/logging.dart';
+
+final _log = Logger('a3::chat::room_description_edit_sheet');
+
+void showEditRoomDescriptionBottomSheet({
+  required BuildContext context,
+  required String description,
+  required String roomId,
+}) {
+  showModalBottomSheet(
+    showDragHandle: false,
+    useSafeArea: true,
+    context: context,
+    isDismissible: false,
+    constraints: const BoxConstraints(maxHeight: 500),
+    builder: (context) {
+      return EditRoomDescriptionSheet(description: description, roomId: roomId);
+    },
+  );
+}
+
+class EditRoomDescriptionSheet extends ConsumerWidget {
+  final String description;
+  final String roomId;
+
+  EditRoomDescriptionSheet({
+    super.key,
+    required this.description,
+    required this.roomId,
+  });
+
+  final _descriptionController = TextEditingController();
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    _descriptionController.text = description;
```

Comment:
this is a side-effect and shouldn't be done in `build` but in `didUpdateDependencies` of a stateful object no? Using a stateless object with a local texteditingcontroller is considered bad style, isn't it? (as it has an internal state but that is hidden to the rendering tree)...

I'd rather you'd make that a proper statefulconsumer and set that not in `build` but have it some of the state-update functions...